src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ToolsApiScannerInstaller.java

```diff
@@ -220,4 +231,19 @@ private void removeOldVersionFileIfExists(File scannerExpansionDirectory) {
         }
     }
 
+
+    private Certificate connectAndGetServerCertificate(HttpUrl httpsServer) {
+        HttpsURLConnection httpsConnection = null;
+        try {
+            httpsConnection = (HttpsURLConnection) httpsServer.url().openConnection();
+            httpsConnection.connect();
+            Certificate[] certificates = httpsConnection.getServerCertificates();
+            httpsConnection.disconnect();
+
+            return certificates[0];
```

Comment:
Any chance of this array being empty? If so, it might be good to check for that.

Reply:
I went with:
```
try {
...
if (certificates.length > 0) {
                return certificates[0];
            } else {
                throw new IOException();
            }
        } catch (IOException e) {
            logger.errorAndDebug("Could not get Black Duck server certificate which is required for managing the local keystore - communicating to the server will have to be configured manually: " + e.getMessage(), e);
            return null;
        } 
```